src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Build/Diagnostics/IncrementalBuildFailureDetector.ProjectChecker.cs

```diff
@@ -49,13 +48,9 @@ private sealed class ProjectChecker : IProjectChecker
             public void OnProjectBuildCompleted()
             {
                 _project.Services.ThreadingPolicy.RunAndForget(
-                    async () =>
-                    {
-                        await TaskScheduler.Default;
-
-                        await CheckAsync(_projectAsynchronousTasksService.UnloadCancellationToken);
-                    },
-                    unconfiguredProject: _project);
+                    () => CheckAsync(_projectAsynchronousTasksService.UnloadCancellationToken),
+                    unconfiguredProject: _project,
+                    options: ForkOptions.StartOnThreadPool | ForkOptions.CancelOnUnload | ForkOptions.NoAssistanceMask);
```

Comment:
What does this one do?

Reply:
Which one do you mean?

- `StartOnMainThread` Ensures that the fork begins execution on the main thread.
- `CancelOnUnload` Allows not executing the delegate if the project context is being disposed of before it would otherwise start execution.
- `NoAssistanceMask` = `SuppressRelevance` | `HideLocks`
- `SuppressRelevance` Any `Microsoft.VisualStudio.Threading.JoinableTask` relevance inherited from the caller is suppressed in the fork.
- `HideLocks` Any project locks not propagated to the fork.